nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp

```diff
@@ -77,15 +77,21 @@ class LifecycleManager : public rclcpp::Node
   void shutdownAllNodes();
   void destroyLifecycleServiceClients();
 
+  // For a node, transition to the new target state
+  bool changeStateForNode(const std::string & node_name, std::uint8_t transition);
+
   // For each node in the map, transition to the new target state
-  void changeStateForAllNodes(std::uint8_t transition);
+  bool changeStateForAllNodes(std::uint8_t transition);
 
   // Convenience function to highlight the output on the console
   void message(const std::string & msg);
 
   // A map of all nodes to be controlled
   std::map<std::string, std::shared_ptr<nav2_util::LifecycleServiceClient>> node_map_;
 
+  // A map of the expected transitions to primary states
+  std::map<std::uint8_t, std::uint8_t> transition_state_map_;
```

Comment:
If there's no benefit from having the elements sorted I'd use a hash table, `unsorted_map`.